client.go

```diff
@@ -57,12 +57,22 @@ type Client struct {
 
 // New creates and returns a new Terraform Enterprise client
 func New(atlasToken string, baseURL string) *Client {
+	return NewWithClient(
+		atlasToken,
+		baseURL,
+		&http.Client{
```

Comment:
any reason not to use a vanilla http client (instead of setting timeout)? Just curious, but this looks great to me regardless!

Reply:
The default timeout is no timeout.  Before I added the client timeout, I'd regularly see requests hange for 5+ minutes, then finally return 500 with empty bodies.

Reply:
^ That sounds about right. But I was just maintaining the old behaviour.

Reply:
oh sorry i didn't realize this was inherited from the previous code!